ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activityEmailInput.component.ts

```diff
@@ -66,63 +73,82 @@ export class ActivityEmailInput implements OnChanges {
         }
     }
 
-    public onChange(answer: string): void {
-        const allFieldsTouched = Object.values(this.emailForm.controls).every(control => control.touched);
-        if (this.emailForm.valid) {
-            this.saveAnswer(answer);
-        } else if (allFieldsTouched) {
-            this.saveAnswer(null);
-        }
-    }
-
     public fieldHasError(field: string, error: string): boolean {
-        return this.emailForm.touched && this.emailForm.controls[field].errors && this.emailForm.controls[field].errors[error];
+        return this.emailForm && this.emailForm.touched && this.emailForm.controls[field].errors
+            && this.emailForm.controls[field].errors[error];
     }
 
     private initEmailForm(): void {
+        interface EmailForm {
+            email: string;
+            confirmEmail?: string;
+        }
+        // can't use the Angular email validator as does not match Pepper back-end email validator
+        const EmailPatternValidator = Validators.pattern(EMAIL_REGEXP);
+
         if (this.block.confirmEntry) {
+
             this.emailForm = this.formBuilder.group({
                 email: new FormControl({
                     value: this.block.answer,
                     disabled: this.readonly
-                }, Validators.email),
+                }, EmailPatternValidator),
```

Comment:
Use our own validator to match back-end email format checker.

Reply:
I'm curious why the email check format doesn't match. AFAIK the server is using a very generous format, basically allowing anything that's not whitespace. If the Angular uses a more stricter format, then it ought to work?

Reply:
The Angular format allows for emails of the form `marco@acme` . Pepper server insists on having the top-level domain, e.g., `marco@acme.com` .

Reply:
Okay, that solves the mystery.